### DISCUSSION TITLE
Separating certain applications for certain users

### DISCUSSION THREAD

**Discussion by tlapik123** [zitadel/zitadel#6859] [Q&A] [answered]
Hi!

I hope this question haven't been asked before. (And I just haven't found it.)

Let's say I have a Gitlab and a Mattermost and I've "connected" them as a application to Zitadel. 
I then have a bunch of users and I want to restrict which users can log-in to which application.

I know that if those were my applications I can just add a bit of code and look for roles and decide on that.
Is it possible for Zitadel to restrict access to some users for some application?
I know Authentik does this, but I'd like to know if Zitadel can do it too. 

Thank you for your time and answers.

**Comment by hifabienne** [comment, marked as answer]
Hei @tlapik123 
You could add a role to the application, e.g user, and then give that role only to the users that are allowed to login.
On the project is a setting, check for authorization. if thats enabled, zitadel will check if the user has any role in the project.

**Reply by tlapik123** [reply]
Thank you for answer.
So if I understand correctly I would have multiple projects - one for gitlab and one for mattermost - with one app in each - and create roles in them + give users those roles. 
Then allow the `Check authorization on Authentication` which would allow only the users with roles in that project.

**Reply by hifabienne** [reply]
yes exactly
